### PR TITLE
allow to disable DevMenu using new method isSecondaryClickToShowDevMenuEnabled

### PR DESCRIPTION
**Pull request by Simek**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

This change allows developers to disable the DevMenu (context menu in macOS case) by ~~reusing currently exported `isShakeToShowDevMenuEnabled` available on iOS~~ addding new `setIsSecondaryClickToShowDevMenuEnabled ` method to the DevSettings:

```
NativeModules.DevSettings.setIsSecondaryClickToShowDevMenuEnabled(false);
```

This also improves quite a bit the development process when using Contextual Menus (aka `ActionSheetIOS`).

Of course I can implement separate, exclusive for macOS, flag if you think it's a better approach.

Also adding `[devSettings isDevModeEnabled]` like in `showOnShake` fixed for me DevMenu appearing on the Release build.

## Changelog

[macOS] [Added] - add ability to disable DevMenu based on new `setIsSecondaryClickToShowDevMenuEnabled` method in DevSettings
[macOS] [Fixed] - disable DevMenu when `isDevModeEnabled` returns `false`

## Test Plan

I have tested the change inside my workbench/sample app, which source code is available [here](https://github.com/Simek/react-native-macos-reminders).

Aug-04-2020 01-38-40

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/528)

### CODE REVIEW

**Comment by tom-un** [comment]
Hi @Simek : Just FYI I've been on vacation and will be on vacation for another week.   Thanks for the contribution.   This is great, but I do think it should have its own DevSetting instead of reusing isShakeToShowDevMenuEnabled.   Perhaps isSecondaryClickToShowDevMenuEnabled.

**Comment by Simek** [comment]
@tom-un Thank you for the review Tom, I have updated the PR according to the request - there is an macOS exclusive method named `isSecondaryClickToShowDevMenuEnabled` now and I have tried to add correct macOS merge comments too. 

Let me know if it's okay or should I update or change anything else. 🙂

**Comment by alloy** [comment]
@tom-un For my understanding, why would it be preferable to add another setting instead of perhaps renaming the existing one, or adding a new generic one and deprecate the existing ‘shake’ one?

**Comment by tom-un** [comment]
> @tom-un For my understanding, why would it be preferable to add another setting instead of perhaps renaming the existing one, or adding a new generic one and deprecate the existing ‘shake’ one?

I just feel its cleaner to have prop names that mean what they say.   Also there could be shake gesture on mac in the future (there already is via Catalyst) or there could be a secondary click gesture in a future iOS.

File: React/DevSupport/RCTDevMenu.m [hunk 1]
```diff
@@ -439,33 +439,38 @@ - (void)setDefaultJSBundle {
...
 - (NSMenu *)menu
 {
-  NSMenu *menu = nil;
-  if (_bridge) {
-    NSString *desc = _bridge.bridgeDescription;
-    if (desc.length == 0) {
-      desc = NSStringFromClass([_bridge class]);
-    }
-    NSString *title = [NSString stringWithFormat:@"React Native: Development\n(%@)", desc];
+  RCTDevSettings *devSettings = _bridge.devSettings; // TODO(OSS Candidate ISS#2710739)
```
Comment (tom-un) [inline React/DevSupport/RCTDevMenu.m:442]: nit: this comment is not necessary: this whole block of code is already enclosed in `#if TARGET_OS_OSX // [TODO(macOS ISS#2323203)` ... `#else // ]TODO(macOS ISS#2323203)`

**Review by tom-un — APPROVED** [review]

**Comment by alloy** [comment]
> I just feel its cleaner to have prop names that mean what they say.

Totally agreed, but I also like to have _less_ API surface for ease of discovery. However…

> Also there could be shake gesture on mac in the future (there already is via Catalyst) or there could be a secondary click gesture in a future iOS.

…I didn’t consider the real possibility of having multiple ways to to open the menu. Although that thinking also makes me wonder if it shouldn’t be an enum?

Anyways, something to consider longer term, I don’t feel strong about it for this PR.

File: React/DevSupport/RCTDevMenu.m (hunk 1 above)
Comment (Simek) [reply React/DevSupport/RCTDevMenu.m:442]: This comment refers to other issue than macOS fork comment, the line has been copied from `showOnShake`:
* https://github.com/microsoft/react-native-macos/blob/5639653093d5b3f02ef64ea9f7440910446e8239/React/DevSupport/RCTDevMenu.m#L184

**Comment by Simek** [comment]
Should I update this PR to 0.62 files structure or should I target the `0.61-stable` branch instead?

**Comment by alloy** [comment]
Update to the 0.62 structure, which is now in `master`, please 🙏

**Comment by Simek** [comment]
> Update to the 0.62 structure, which is now in `master`, please 🙏

Not a problem @alloy, I will work on the changes this week 🙂

**Comment by Simek** [comment]
@alloy Should I force push the branch again to rerun the `ci/circleci: test_js`? It looks like this failure is not related to the changes in this PR.

**Comment by tom-un** [comment]
> @alloy Should I force push the branch again to rerun the `ci/circleci: test_js`? It looks like this failure is not related to the changes in this PR.

I tried forcing it to re-run and it failed again.   I'm not sure why.   As a branch of the upstream react-native we get these circleci tests too.   They're mostly redundant as we do all the same jest, lint, flow and format-check tests in Azure DevOps.   I'll force merge this PR and watch for failures in other PRs.

Thanks!

**Comment by HeyImChris** [comment]
> > @alloy Should I force push the branch again to rerun the `ci/circleci: test_js`? It looks like this failure is not related to the changes in this PR.
> 
> I tried forcing it to re-run and it failed again. I'm not sure why. As a branch of the upstream react-native we get these circleci tests too. They're mostly redundant as we do all the same jest, lint, flow and format-check tests in Azure DevOps. I'll force merge this PR and watch for failures in other PRs.
> 
> Thanks!

Late here but they're really flaky. I wouldn't go off of the circle CI tests at all as long as our ADO tests pass.

**Comment by alloy** [comment]
@Simek Thanks again for your work! 🙌

**Pull request merged**